zerver/views/__init__.py

```diff
@@ -874,6 +874,7 @@ def home(request):
         fullname              = user_profile.full_name,
         email                 = user_profile.email,
         domain                = user_profile.realm.domain,
+        domains               = list_of_domains_for_realm(user_profile.realm),
```

Comment:
I guess this overall approach works and will look reasonably nice for the display code, but I would probably have just change the checkbox to say "Restrict new users to certain domains" and then have a table of the actual domains below it that expands out, rather than marshalling things into a string.